CMake: sync with new jrl-cmakemodules

nim65s opened the pull request.

nim65s committed:
sync submodule

nim65s committed:
CMake: sync with new jrl-cmakemodules

nim65s committed:
package.xml: updates

nim65s commented:
I have no more ideas how I've been able to break this on windows:
```

1: Test command: C:\Miniconda\envs\pycppad\python.exe "D:/a/pycppad/pycppad/example/add_eq.py"
1: Working Directory: D:/a/pycppad/pycppad/build/example
1: Environment variables: 
1:  PYTHONPATH=D:/a/pycppad/pycppad/build/python
1: Test timeout computed to be: 1500
1: Traceback (most recent call last):
1:   File "D:/a/pycppad/pycppad/example/add_eq.py", line 3, in <module>
1:     from pycppad import AD, Independent, ADFun, Value
1: ImportError: cannot import name 'AD' from 'pycppad' (unknown location)
1/1 Test #1: pycppad-add_eq ...................***Failed    0.03 sec
Traceback (most recent call last):
  File "D:/a/pycppad/pycppad/example/add_eq.py", line 3, in <module>
    from pycppad import AD, Independent, ADFun, Value
ImportError: cannot import name 'AD' from 'pycppad' (unknown location)

```

@jorisv : any clue ?

(The last commit should probably be discarded)

jorisv commented:
Hello @nim65s,
I checked previous log and this test was failing without making the CI fail.
I will have a look.

jorisv commented:
Apparently, an error in CMakeLists.txt forced us to fetch content the jrl-cmakemodule.
This create an error in [where wrong doxyfile path](https://github.com/jrl-umi3218/jrl-cmakemodules/blob/master/doxygen.cmake#L546) was found.
Conda CI was using github image doxygen that have an inconsistent output on this error.

jorisv committed:
ci: Modernize CI and update dependencies

- Avoid to use default channel
- Use conda clang-cl
- Use conda doxygen
- Use most recent version of Boost

jorisv committed:
cmake: Fix jrl-cmake-modules include

jorisv committed:
cmake: Fix running test in build directory on Windows

jorisv commented:
All is fixed on my side
